mobile/lib/shared/views/backup_controller_page.dart

```diff
@@ -89,13 +120,36 @@ class BackupControllerPage extends HookConsumerWidget {
                   : Container(),
               Padding(
                 padding: const EdgeInsets.only(top: 8.0),
-                child: OutlinedButton(
-                  onPressed: () {
-                    isAutoBackup
-                        ? ref.watch(authenticationProvider.notifier).setAutoBackup(false)
-                        : ref.watch(authenticationProvider.notifier).setAutoBackup(true);
-                  },
-                  child: Text("Turn $backupBtnText Backup", style: const TextStyle(fontWeight: FontWeight.bold)),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    DropdownButton<String>(
```

Comment:
I would split this DropdownButton to a separate function for cleaner code and easier to put conditional statements in

```dart
    _buildAlbumSelectDropDownMenu() {
      if (assetEntities.hasData) {
        List<AssetPathEntity>? albums = assetEntities.data;

        if (albums != null) {
          return DropdownButton<String>(
            hint: const Text("Select album to backup"),
            icon: const Icon(Icons.photo),
            isExpanded: true,
            items: (albums
                .map(
                  (AssetPathEntity album) => {
                    DropdownMenuItem(
                      value: album.id,
                      child: Text(album.name),
                    )
                  }.first,
                )
                .toList()),
            value: backupAlbumId.isEmpty ? albums[0].id : backupAlbumId,
            onChanged: (albumId) {
              if (albumId != null) saveAlbumId(albumId);
            },
          );
        }
      }

      return Container();
    }
```

Reply:
Thanks, I didn't consider not building the widget if the albums are not loaded yet. This way it's much easier, thanks again!